### DISCUSSION TITLE
Conditional Form Validation

### DISCUSSION THREAD

**Discussion by mpestanasoares** [filamentphp/filament#4129] [General]
Let's say we have a large form and there are two options: save it for later and send it permanently. In the first case, validation rules can be ignored, but not in the second. Is there a way to make validation conditional in Filament Forms?

**Comment by danharrin** [comment]
Each validation rule can be conditional. For example:

`required(fn () => $this->shouldValidateForm)`
`maxLength(fn () => $this->shouldValidateForm ? 255 : null)`

Then on your Livewire component:

```php
protected bool $shouldValidateForm = true;

public function save(bool $andSend = true)
{
    $this->shouldValidateForm = $andSend;
    
    $data = $this->form->getState();
    
    // ...
    
    $this->shouldValidateForm = true;
}
```

**Reply by mpestanasoares** [reply]
got it!
